formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Encoding.kt

```diff
@@ -392,20 +656,67 @@ internal class CborDecoder(private val input: ByteArrayInput) {
             input.readExactNBytes(strLen)
         }
 
-    private fun skipOverTags() {
+    private fun processTags(tags: ULongArray?): ULongArray? {
+        val collectedTags = mutableListOf<ULong>()
         while ((curByte and 0b111_00000) == HEADER_TAG) {
-            readNumber() // This is the tag number
+            val readTag = readNumber().toULong() // This is the tag number
+            collectedTags += readTag
             readByte()
         }
+
+        return (if (collectedTags.isEmpty()) null else collectedTags.toULongArray()).also { collected ->
```

Comment:
IMO `also` creates unnecessary nesting here.

Reply:
it needs to be transformed into an `ULongArray` for comparison *and* for returning it. the alternative of adding a temp variable just for two accesses seems a bit cumbersome to me